solutions/devsprint-riccardo-1/OnboardingChallenge/Screens/Settings/Views/SettingsViewController.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  SettingsViewController.swift
+//  OnboardingChallenge
+//
+//  Created by Rodrigo Borges on 03/11/21.
+//
+
+import UIKit
+
+let NUMBEROFROWSINSECTION = 1
+
+protocol SettingsViewDisplayLogic {
+    func display(viewModel: SettingsViewConfiguration.Get.ViewModel)
+}
+
+class SettingsViewController: UIViewController {
+    
+    private lazy var settingsView: SettingsView = {
+        return SettingsView()
+    }()
+    
+    var interactor: SettingsBusinessLogic?
+    var presenter: SettingsPresentationLogic?
+        
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        let request = SettingsViewConfiguration.Get.Request(key: "CFBundleVersion")
+        interactor?.getAppVersion(request: request)
```

Comment:
I think that this is part of the pattern, but what happens if we forget about asking the interaction to get the App version? The view won't present anything on the screen and this looks suspicious: we don't want to explicitly ask our view to do something. We are human and we have many things going on: it's easy to forget to add a line of code.
A better approach is to give the compiler the responsibility to remember us about that.
A way to do that, is by implementing an initializer for the `SettingView` that requires us the app version (or the sections we want to render). 
By requesting that, we need to find a way to retrieve the app version. The interaction can do that for us, so we instantiate an interaction and request the app version.

Try to implement these changes: you will see how the compiler drive the implementation and it also helps us not forget about anything.

Reply:
This has not been fixed properly, however there has been some cleanups. I am not sure how to implement the logic whereby the viewModel loads the data on load.